src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -199,7 +199,7 @@ private static function doRemove(array $files, bool $isRecursive): void
 
                     throw new IOException(sprintf('Failed to remove directory "%s": ', $file).$lastError);
                 }
-            } elseif (!self::box('unlink', $file) && (str_contains(self::$lastError, 'Permission denied') || file_exists($file))) {
+            } elseif (!self::box('unlink', $file) && (str_contains((string) self::$lastError, 'Permission denied') || file_exists($file))) {
```

Comment:
I'd rather use `self::$lastError && str_contains(self::$lastError, 'Permission denied')` here to make the intent easier to understand here (handling the case where there is no last error)